### ISSUE TITLE
LaTeX reader transforms math environments preventing HTML+MathJax equation numbering

### ISSUE THREAD

**Issue by fedeinthemix** [jgm/pandoc#7508]
I would like to convert a LaTeX document into HTML and get equations numbered. MathJax supports automatic equation numbering and supports most AMS math environments (http://docs.mathjax.org/en/latest/input/tex/eqnumbers.html).

The problem I'm having is that `pandoc ` seems to convert all (?) math environments to displaymath delimited by `\[ ... \]`. However, the latter notation doesn't produce any numbering. For example the following LaTeX document

```latex
\documentclass{article}

\usepackage{amsmath}

\begin{document}
The following equation
\begin{equation}
  \label{eq:energy}
  E = m c^2
\end{equation}
is correctly numbered and can be referred to: Eq.~(\ref{eq:energy}), Eq.~\eqref{eq:energy}.
\end{document}
```
is converted by the command `pandoc -t html --mathjax toHtmlMathJax.tex |m` into

```html
<p>The following equation <span class="math display">\[\label{eq:energy}
  E = m c^2\]</span> is correctly numbered and can be referred to: Eq. (<a href="#eq:energy" data-reference-type="ref" data-reference="eq:energy">[eq:energy]</a>), Eq. <a href="#eq:energy" data-reference-type="eqref" data-reference="eq:energy">[eq:energy]</a>.</p>
```

If I manually replace the delimiters `\[` and `\]` with the original `\begin{equation}`, `\end{equation}` and use a template to activate MathJaX numbering, then numbering works as expected.

For markdown to html conversion [here](https://groups.google.com/g/pandoc-discuss/c/bW9DW5nQy50) is suggested to embed the `equation` environment in `\[...\]`.  However, this results in and invalid LaTeX document which is my golden document and used directly for PDF output.

It would be great if the `pandoc` LaTeX reader could keep the various math environments. 

I've checked with `pandoc` 2.14.1.

**Comment by jgm** [comment]
See #4104 which explains the issues around this and suggests a workaround.

We mention there a possible change to pandoc

> One good solution might be to modify mathEnvWith in the LaTeX reader so that, if the raw_tex extension is enabled, these environments are parsed as raw latex; otherwise, we do as before and parse as math with necessary modifications.

However, as noted:

> Since raw_tex is enabled by default in pandoc markdown, this might mean that some existing documents would break on conversion to Word, so that's a potential worry.

If we were only supporting output to HTML and LaTeX, this would be trivial.  But we need to give good-as-possible math conversions to a large number of formats.

**Comment by jgm** [comment]
I don't think this issue adds anything to #4014 so it should probably be closed as a dup.

**Comment by fedeinthemix** [comment]
I missed #4104 and agree that its essentially the same issue.

In any case I would like to correct one point of what I wrote above: in the generated HTML the `<a>...</a>` links to the equations generated from `\ref{eq:energy}` and `\eqref{eq:energy}` show up as `[eq:energy]`. For this reason MathJax doesn't recognize them and do not transform them in the corresponding equation number. For MathJax to do its work, references to equation numbers should be left as `\ref` resp. `\eqref`.

**Comment by tarleb** [comment]
Closing in favor of #4104 as noted above.
